[CI] CreateIndexLimitIT testCreateIndexLimit failing

**Build Scans:**
- [elasticsearch-periodic #12212 / release-tests](https://gradle-enterprise.elastic.co/s/nsfrw426s3ydo)
- [elasticsearch-periodic #12212 / release-tests](https://gradle-enterprise.elastic.co/s/lbhf3hd43m2x4)
- [elasticsearch-pull-request #142304 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/r55ddqdpbfapy)
- [elasticsearch-pull-request #142304 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/xmn3rtbds5obk)
- [elasticsearch-periodic-java-ea #254 / release-tests](https://gradle-enterprise.elastic.co/s/j5r6xe3t2m6us)
- [elasticsearch-pull-request #142124 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/zup4ycyvxcsou)
- [elasticsearch-pull-request #142124 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/ousrbikbvgsgc)
- [elasticsearch-pull-request #142040 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/pysnp5yydqv4k)
- [elasticsearch-periodic-java-ea #253 / release-tests](https://gradle-enterprise.elastic.co/s/7lzyf6nfoiwv4)
- [elasticsearch-periodic #12202 / release-tests](https://gradle-enterprise.elastic.co/s/dow2e42opty7c)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.cluster.metadata.CreateIndexLimitIT.testCreateIndexLimit" -Dtests.seed=3A1F29E42BD12D6B -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=doi -Dtests.timezone=Asia/Almaty -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.cluster.metadata.CreateIndexLimitIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testCreateIndexLimit'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a string containing "see https://www.elastic.co/docs/deploy-manage/deploy/elastic-cloud/differences-from-other-elasticsearch-offerings.md?version=master#index-and-resource-limits"
     but: was "This action would add an index, but this project currently has [3]/[3] maximum indices; for more information, see https://www.elastic.co/docs/deploy-manage/deploy/elastic-cloud/differences-from-other-elasticsearch-offerings.md?version=9.5#index-and-resource-limits"
```

**Issue Reasons:**
- [main] 6 consecutive failures in step release-tests
- [main] 5 consecutive failures in step checkpart2_release-tests
- [main] 11 failures in test testCreateIndexLimit (2.2% fail rate in 509 executions)
- [main] 6 failures in step release-tests (100.0% fail rate in 6 executions)
- [main] 5 failures in step checkpart2_release-tests (100.0% fail rate in 5 executions)
- [main] 2 failures in pipeline elasticsearch-periodic (66.7% fail rate in 3 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (1.1% fail rate in 285 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-java-ea (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -515,9 +515,6 @@ tests:
 - class: org.elasticsearch.xpack.searchablesnapshots.store.input.CachedBlobContainerIndexInputTests
   method: testRandomReads
   issue: https://github.com/elastic/elasticsearch/issues/148163
-- class: org.elasticsearch.xpack.stateless.cluster.metadata.CreateIndexLimitIT
-  method: testCreateIndexLimit
-  issue: https://github.com/elastic/elasticsearch/issues/147956
 - class: org.elasticsearch.xpack.stateless.recovery.SearchShardRelocationIT
   method: testUnpromotableRelocationHandoffFailsIfTargetAllocationIdChanges
   issue: https://github.com/elastic/elasticsearch/issues/147957

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/cluster/metadata/CreateIndexLimitIT.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/cluster/metadata/CreateIndexLimitIT.java
@@ -23,6 +23,7 @@ import org.elasticsearch.cluster.metadata.MetadataCreateIndexService;
 import org.elasticsearch.cluster.metadata.Template;
 import org.elasticsearch.cluster.project.ProjectResolver;
 import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.ReferenceDocs;
 import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.compress.CompressedXContent;
 import org.elasticsearch.common.settings.Settings;
@@ -244,13 +245,7 @@ public class CreateIndexLimitIT extends AbstractStatelessPluginIntegTestCase {
                 IndexLimitExceededException.class,
                 prepareCreate(indexPattern + suffix)
             );
-            assertThat(
-                e.getMessage(),
-                containsString(
-                    "see https://www.elastic.co/docs/deploy-manage/deploy/"
-                        + "elastic-cloud/differences-from-other-elasticsearch-offerings.md?version=master#index-and-resource-limits"
-                )
-            );
+            assertThat(e.getMessage(), containsString("see " + ReferenceDocs.MAX_INDICES_PER_PROJECT));
         }
     }
 

